src/app/pages/sharing/smb/smb-form/smb-name-validator.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import {
+  AbstractControl, ValidationErrors,
+} from '@angular/forms';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  Observable, catchError, debounceTime, distinctUntilChanged, of, switchMap, take,
+} from 'rxjs';
+import { WebSocketService } from 'app/services/ws.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SmbNameValidationService {
+  constructor(
+    private ws: WebSocketService,
+    private translate: TranslateService,
+  ) { }
+
+  private noSmbUsersError = this.translate.instant('TrueNAS server must be joined to Active Directory or have at least one local SMB user before creating an SMB share');
+  private nameExistsError = this.translate.instant('Share with this name already exists');
+
+  private errorText: string;
+
+  validateSmbName = (control: AbstractControl, existingName: string): Observable<ValidationErrors | null> => {
```

Comment:
I've noticed that naming of this function is about validation of the name, while the function is responsible for more than that - it's prepared to be responsible for handling any other error, including the "smbUsersError".

I'd suggested a concise self-documenting naming for this function, for example "validate"